zict/sieve.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import absolute_import, division, print_function
+
+from collections import MutableMapping
+from itertools import chain
+import sys
+
+
+class Sieve(MutableMapping):
+    """ Store values in different mappings based on their sizes.
+
+    This creates a MutableMapping by combining two MutableMappings,
+    one that stores values strictly smaller (in bytes) than a given
+    threshold, the other storing the other values.
+
+    Only the size of values, not keys, is considered, since keys are
+    likely kept in memory for bookkeeping purposes anyway.
+
+    Parameters
+    ----------
+    small: MutableMapping
+    large: MutableMapping
+    threshold: int
+        The number of bytes below which values are stored in the *small*
+        mapping.
+
+    Examples
+    --------
+    >>> small = {}
+    >>> large = Func(dumps, loads, File('storage/'))  # doctest: +SKIP
+    >>> store = Sieve(small, large, threshold=1024 ** 2)  # doctest: +SKIP
+
+    See Also
+    --------
+    Buffer
+    """
+    def __init__(self, small, large, threshold):
```

Comment:
In that case, why not simply let the whole decision up to the function? ie make it return True/False for small/large. Personally, I would even generalize this into a collection (list or dict, whatever the user passes) of zict and a decision function which returns the indexer of the zict to use, so that you can have sharding!